### DISCUSSION TITLE
Is there are way to import a js module hosted on github?

### DISCUSSION THREAD

**Discussion by solidsnail** [community/community#25343] [Other Feature Feedback, Questions, & Ideas]
<p>Hi there,</p>
<p>I got a js module hosted on a repo, I try to import it in my project like so :</p>
<pre><code class="lang-auto">import "https://raw.githubusercontent.com/solidsnail/reponame/master/path/to/index.js"
</code></pre>
<p>But apparently github returns the file as a raw text ( Content-Type: text/plain ), I want it to return text/javascript instead, is this somehow possible ? Because I would love to host my modules on github instead of cdn services.</p>
<p>Thanks in advance</p>

**Comment by clarkbw** [comment]
<p>GitHub doesn’t offer a way to do this right now.  Services like unpkg do this for npm packages and there are other services that do this as well but not official GitHub services.</p>

**Comment by solidsnail** [comment]
<p>Any plans on supporting this in the future or is this out of the scope of Github goals ?</p>

**Comment by clarkbw** [comment]
<p>We’re interested in doing this to support <a href="https://deno.land/" rel="nofollow noopener">Deno</a> and other use cases but the timeline is unclear.</p>

**Comment by jcansdale** [comment, marked as answer]
<p>Using GitHub Pages appears to be a popular workaround, see:<br>
</p><aside class="onebox stackexchange">
  <header class="source">
      <a href="https://stackoverflow.com/questions/8779197/how-to-link-files-directly-from-github-raw-github-com/50598749#50598749" target="_blank" rel="nofollow noopener">stackoverflow.com</a>
  </header>
  <article class="onebox-body">
      <a href="https://stackoverflow.com/users/7919171/sista-fiolen" target="_blank" rel="nofollow noopener">
    <img alt="Sista Fiolen" src="https://user-images.githubusercontent.com/11719160/181090536-10864689-c82f-4206-b377-712c1bacbbcd.jpeg" class="thumbnail onebox-avatar" width="128" height="128">
  </a>
<h4>
  <a href="https://stackoverflow.com/questions/8779197/how-to-link-files-directly-from-github-raw-github-com/50598749#50598749" target="_blank" rel="nofollow noopener">How to link files directly from Github (raw.github.com)</a>
</h4>

<div class="tags">
  <strong>github, github-pages, cdn</strong>
</div>

<div class="date">
  
  answered by
  <a href="https://stackoverflow.com/users/7919171/sista-fiolen" target="_blank" rel="nofollow noopener">
    Sista Fiolen
  </a>
  on <a href="https://stackoverflow.com/questions/8779197/how-to-link-files-directly-from-github-raw-github-com/50598749#50598749" target="_blank" rel="nofollow noopener">07:26AM - 30 May 18 UTC</a>
</div>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by solidsnail** [comment]
<p>Works perfectly ! Thanks y’all</p>
